omc3/kmod/analysis.py

```diff
@@ -217,15 +220,18 @@ def do_fit(magnet_df, plane, use_approx=False):
     if not np.any(sigma):
         sigma = 1.E-22 * np.ones(len(sigma))
 
-    av_beta, av_beta_err = scipy.optimize.curve_fit(
-        fun,
-        xdata=return_fit_input(magnet_df, plane),
-        ydata=magnet_df.where(magnet_df[f"{CLEANED}{plane}"])[
-            f"{TUNE}{plane}"].dropna() - magnet_df.headers[f"{TUNE}{plane}"],
-        sigma=sigma,
-        absolute_sigma=True,
-        p0=1
-    )
+    # We filter out the "Covariance of the parameters could not be estimated" warning
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=OptimizeWarning)
+        av_beta, av_beta_err = scipy.optimize.curve_fit(
+            fun,
+            xdata=return_fit_input(magnet_df, plane),
+            ydata=magnet_df.where(magnet_df[f"{CLEANED}{plane}"])[
+                f"{TUNE}{plane}"].dropna() - magnet_df.headers[f"{TUNE}{plane}"],
+            sigma=sigma,
+            absolute_sigma=True,
+            p0=1
+        )
```

Comment:
**Note**: I have filtered the `OptimizeWarning: Covariance of the parameters could not be estimated` one as it feels like we can't do much about it, but we might decide to keep them.

Reply:
I would leave it in, as it gives the user some feedback that there might be an issue and how much to trust the fit!

Reply:
In this case, I would like to catch the warnings anyway but emit it as a log message ourselves. This way we are not polluting `stderr` and the user can choose to see this or not by setting the logging level. Thoughts?